src/Codeception/Lib/InnerBrowser.php

```diff
@@ -507,17 +523,20 @@ private function clickButton(\DOMNode $node)
         throw new TestRuntimeException('Button is not inside a link or a form');
     }
 
-    private function openHrefFromDomNode(\DOMNode $node)
+    private function openHrefFromDomNode(DOMNode $node)
     {
         $link = new Link($node, $this->getBaseUrl());
-        $this->amOnPage(preg_replace('/#.*/', '', $link->getUri()));
+        $this->amOnPage(preg_replace('#\#.*#', '', $link->getUri()));
```

Comment:
This is the only case when `#` is worse pattern delimiter than `/`. How about changing it back?